client/Dockerfile

```
@@ -3,6 +3,12 @@
 # ============================================
 FROM oven/bun:1.2.22-alpine AS dev
 
+# Install Node.js to run the dev server. Bun's JSC engine has known memory
+# retention under long-running Next.js dev loads (oven-sh/bun#27514, #29302),
+# causing OOM-kills in Docker Desktop. Bun is still used for `install` because
+# the leak only matters for long-lived processes.
+RUN apk add --no-cache nodejs npm
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)